xarray/core/computation.py

```diff
@@ -927,45 +927,48 @@ def earth_mover_distance(first_samples,
 
 
 def dot(*arrays, **kwargs):
-    """ dot(*arrays, *, dims=None)
+    """ dot(*arrays, dims=None)
 
-    einsum for xarray object, but providing simpler interface based on
-    the array dimensions.
+    Generalized dot product for xarray objects. Like np.einsum, but
+    provides a simpler interface based on array dimensions.
 
     Parameters
     ----------
-    arrays: multiple DataArrays
-        arrays to compute.
-    dims: tuple of strings, optional
-        Along which dimensions to be summed over.
+    arrays: DataArray objects
+        Arrays to compute.
+    dims: str or tuple of strings, optional
+        Which dimensions to sum over.
         If not speciified, then all the common dimensions are summed over.
 
     Returns
     -------
-    dot: same type to input.
+    dot: DataArray
 
     Examples
     --------
 
     >>> da_a = xr.DataArray(np.arange(3 * 4).reshape(3, 4), dims=['a', 'b'])
     >>> da_b = xr.DataArray(np.arange(3 * 4 * 5).reshape(3, 4, 5),
-                            dims=['a', 'b', 'c'])
+    >>>                     dims=['a', 'b', 'c'])
     >>> da_c = xr.DataArray(np.arange(5 * 6).reshape(5, 6), dims=['c', 'd'])
-
-    >>> dot(da_a, da_b, dims=['a', 'b']).dims
+    >>>
+    >>> xr.dot(da_a, da_b, dims=['a', 'b']).dims
     ('c', )
-    >>> dot(da_a, da_b, dims=['a']).dims
+    >>> xr.dot(da_a, da_b, dims=['a']).dims
     ('b', 'c')
-    >>> dot(da_a, da_b, da_c, dims=['b', 'c']).dims
+    >>> xr.dot(da_a, da_b, da_c, dims=['b', 'c']).dims
     ('a', 'd')
     """
     from .dataarray import DataArray
     from .variable import Variable
 
     dims = kwargs.pop('dims', None)
+    if len(kwargs) > 0:
+        raise TypeError('Invalid keyward arguments {} are given'.format(
+            list(kwargs.keys())))
 
-    if len(arrays) < 2:
-        raise TypeError('More than one arrays must be provided')
+    if len(arrays) < 2 and dims is None:
+        raise TypeError('dim must be provided for one array computation.')
```

Comment:
If there's only one array, wouldn't `dims` just be any repeated dimensions on the single array?